src/registries/repeat_commands_registry.ts

```diff
@@ -0,0 +1,149 @@
+
+repeatCommandTransformRegistry.add("ADD_MERGE", undefined);
+repeatCommandTransformRegistry.add("REMOVE_MERGE", undefined);
+
+repeatCommandTransformRegistry.add("SET_FORMATTING", undefined);
+repeatCommandTransformRegistry.add("CLEAR_FORMATTING", undefined);
+repeatCommandTransformRegistry.add("SET_BORDER", undefined);
+
+repeatCommandTransformRegistry.add("CREATE_FILTER_TABLE", undefined);
+repeatCommandTransformRegistry.add("REMOVE_FILTER_TABLE", undefined);
+
+repeatCommandTransformRegistry.add("HIDE_SHEET", undefined);
+
+repeatCommandTransformRegistry.add("ADD_COLUMNS_ROWS", repeatAddColumnsRowsCommand);
+repeatCommandTransformRegistry.add("REMOVE_COLUMNS_ROWS", repeatHeaderElementCommand);
+repeatCommandTransformRegistry.add("HIDE_COLUMNS_ROWS", repeatHeaderElementCommand);
+repeatCommandTransformRegistry.add("RESIZE_COLUMNS_ROWS", repeatHeaderElementCommand);
+
+repeatCommandTransformRegistry.add("CREATE_SHEET", repeatCreateSheetCommand);
+repeatCommandTransformRegistry.add("DUPLICATE_SHEET", repeatDuplicateSheetCommand);
+
+repeatCommandTransformRegistry.add("CREATE_FIGURE", repeatCreateFigureCommand);
+repeatCommandTransformRegistry.add("CREATE_CHART", repeatCreateChartCommand);
+repeatCommandTransformRegistry.add("CREATE_IMAGE", repeatCreateImageCommand);
+
+export const repeatableLocalCommandsRegistry = new Registry<LocalRepeatTransform | undefined>();
+repeatableLocalCommandsRegistry.add("STOP_EDITION", getRepeatedLocalCommandChildrenGeneric);
+repeatableLocalCommandsRegistry.add("PASTE", repeatPasteCommand);
+repeatableLocalCommandsRegistry.add("INSERT_CELL", repeatInsertOrDeleteCellCommand);
+repeatableLocalCommandsRegistry.add("DELETE_CELL", repeatInsertOrDeleteCellCommand);
+repeatableLocalCommandsRegistry.add("AUTORESIZE_COLUMNS", repeatAutoresizeCommand);
+repeatableLocalCommandsRegistry.add("AUTORESIZE_ROWS", repeatAutoresizeCommand);
+repeatableLocalCommandsRegistry.add("SORT_CELLS", repeatSortCellsCommand);
+repeatableLocalCommandsRegistry.add("SUM_SELECTION", getRepeatedLocalCommandGeneric);
+repeatableLocalCommandsRegistry.add("SET_DECIMAL", getRepeatedLocalCommandGeneric);
+
+export function genericTransform<T extends Command>(
+  getters: Getters,
+  command: T,
+  basePosition?: Position,
+  mode?: "standalone" | "grouped"
+): T {
+  let transformedCommand = deepCopy(command);
+
+  for (const transform of genericTransforms) {
+    transformedCommand = transform(getters, transformedCommand);
+  }
+
+  return transformedCommand;
+}
+
+/** Apply generic transforms (sheetID, target, ...) to a local command */
+export function getRepeatedLocalCommandGeneric<T extends LocalCommand>(
+  getters: Getters,
+  cmd: T,
+  childCommands: readonly CoreCommand[]
+): T | undefined {
+  return genericTransform(getters, cmd);
+}
+
+/** Apply generic transforms (sheetID, target, ...) to the core command children of a local command */
+export function getRepeatedLocalCommandChildrenGeneric<T extends LocalCommand>(
```

Comment:
this function is only used once for STOP_EDITION. maybe we could just move (and rename) this to a specific repeat function `repeatStopEdition`. It would be easier to understand than this kinda barbaric function name

Reply:
If I don't need to use it as a generic for odoo commands, I'll do that yeah.

Reply:
Looks like we don't need that in odoo, changed it to `repeatStopEdition`

Reply:
Edit: cannot make it into a specific transform because of cyclic imports. `repeat_commands_registry.ts` need `repeatStopEdition` to put in the registry => `repeatStopEdition` need `repeatCoreCommand` => `repeatCoreCommand` need the registry. 

I put it back in this file, but changed the name to `repeatLocalCommandChildren` which is elss of a mouthful